docs/query_execution_plan_in_TiSpark.md

```diff
@@ -0,0 +1,84 @@
+
+## Understand EXPLAIN output in `CoprocessorRDD`(`FetchHandleRDD`)
+
+### Operator that perform data scans
+
+An operator is a particular step that is executed as part of returning query results. The operators that perform data scans (of the disk or the TiKV/TiFlash Block Cache) are listed as follows:
+
+* **`TableRangeScan`**: Table scans with the specified range. We consider full table scan as a special case of `TableRangeScan`, so full table scan is also called `TableRangeScan`.
+* **`TableRowIDScan`**: Scans the table data based on the `RowID`. Usually follows an index read operation to retrieve the matching data rows.
+* **`IndexRangeScan`**: Index scans with the specified range. We consider full index scan as a special case of `IndexRangeScan`, so full index scan is also called `IndexRangeScan`.
+
+### Operator that aggregates the data from TiKV/TiFlash
+
+TiSpark aggregates the data or calculation results scanned from TiKV/TiFlash. The data aggregation operators can be divided into the following categories:
+
+- **`TableReader`**: Aggregates the data obtained by the underlying operator `TableRangeScan` in TiKV/TiFlash.
+- **`IndexReader`**: Aggregates the data obtained by the underlying operator `IndexRangeScan` in TiKV/TiFlash.
+- **`IndexLookUp`**: First aggregates the RowIDs (in TiKV/TiFlash) scanned by the first scan in the index. Then at the second scan in the table, accurately reads the data from TiKV/TiFlash based on these RowIDs. At the first scan in the index, there is `IndexRangeScan` operator; at the second scan in the table, there is the `TableRowIDScan` operator.
+
+### `Range`&`RangeFilter`
+
+In the `WHERE`/`HAVING`/`ON` conditions, the TiSpark optimizer analyzes the result returned by the primary key query or the index key query. For example, these conditions might include comparison operators of the numeric and date type, such as `>`, `<`, `=`, `>=`, `<=`, and the character type such as `LIKE`.
+
+The `Range` in `CoprocessorRDD`(`FetchHandleRDD`) represents the range of scanning. `RangeFilter` indicates which expression(s) the range is made up of. If `RangeFilter` is empty, it indicates a full table scan or full index scan. `RangeFilter` generally appears when the query involves an index range, when query the expressions in the `RangeFilter` form the scanned range from left to right.
+
+### Selection
+
+The expression passed to COP/TiKV as selection expression without triggering a downgrade.
+
+## Understand EXPLAIN output in `RegionTaskExec`
+
+From the previous article, we know that `IndexLookUp` will perform two scanning operations, the first scan is `IndexRangeScan` and the second scan is `TableRowIDScan`. If the `TableRowIDScan` in `IndexLookUp` does too many queries on COP([TiKV Coprocessor](https://docs.pingcap.com/tidb/stable/tikv-overview#tikv-coprocessor))/TiKV, it can cause performance problems in COP/TiKV. To solve this problem a downgrading mechanism is introduced.
```

Comment:
```suggestion
From the previous section, we know that `IndexLookUp` will perform two scanning operations, the first scan is `IndexRangeScan` and the second scan is `TableRowIDScan`. If the `TableRowIDScan` in `IndexLookUp` does too many queries on COP([TiKV Coprocessor](https://docs.pingcap.com/tidb/stable/tikv-overview#tikv-coprocessor))/TiKV, it can cause performance problems in COP/TiKV. To solve this problem a downgrading mechanism is introduced.
```

Reply:
changed